### ISSUE TITLE
Control throwing exceptions when executing python code in Output

### ISSUE THREAD

**Issue by sok82** [datalayer/jupyter-ui#301]
### Problem
When python code executed in `Output` contains errors, KernelExecutor throws Javascript exception that can't be caught or processed in application using `Output`.

Usually it's quite a normal thing during development when some code throws an exception and application where we develop code must not crash just because executed code is not ok yet.

Moreover,  necessary information about exception that could be helpful for user is already present in execution handlers \ model listeners and having uncontrolled exception seems like overkill

### Proposed Solution

Introduce new parameter in `Output` that will control if exception must be thrown, by default do not change current 

### Additional context
For now it works like this

We render `Output` component on page
```ReactJS
  <div style={{padding: cell && cell.output && !errorData ? '5px' : '', maxWidth:CELL_CONTENT_MAX_WIDTH}}>
                    <Output                    
                        id={outputId}
                        codePre=""
                        autoRun={false}
                        kernel={props.kernel}
                        code={
                            props.code && typeof props.code === 'string' 
                                ? props.code
                                : ''
                        }
                        showEditor={false}
                        showControl={false}
                        showKernelProgressBar={false}                                                
                        executeTrigger={executeTrigger}      
                    />
</div>    
```
and trigger code execution by incrementing `executeTrigger` parameter

And if `code` contains errors we see that `KernelExecutor` that is called by underlying `OutputExecutor` throws
exception like

```
Uncaught (in promise) <Error description>
```
![image](https://github.com/user-attachments/assets/9ee52925-ebd3-409e-a67b-a6f8afc00043)

It looks like this behaviour was introduced in versions > 0.14, in older versions this was not an issue.

I can possibly create PR to fix this 

cc@echarles

**Comment by echarles** [comment]
@sok82 I think (not sure) I face such issue and brought a fix recently. Could you check with latest main the following examples

https://github.com/datalayer/jupyter-ui/blob/main/packages/react/src/examples/KernelExecutor.tsx
https://github.com/datalayer/jupyter-ui/blob/main/packages/react/src/examples/KernelExecute.tsx

I you type some garbage in the KernelExecutor example, it should not be shown correctly. If not, or if you issue addresses another case, happy to review a PR.

**Comment by sok82** [comment]
@echarles I just checked this scenario on `0.18.4` and still having this issue with uncaught runtime exception in case of code with errors

**Comment by echarles** [comment]
The error is logged in the devconsole, but can be shown in the UI, as done in the KernelExcecute example

![image](https://github.com/user-attachments/assets/3e0f01cd-11f9-450c-9374-6c8c0ad6b80f)

But you may have another scenario/requirement, feel free to open a PR.

**Comment by echarles** [comment]
```
    kernel?.execute(code)?.result
      .then((result: string) => {
        setResult(result);
      })
      .catch((error: string) => {
        console.log('Error', error);
        setResult(error);
      })
      .finally(() => {
        setRunning(false);  
      }
    );
```

**Comment by sok82** [comment]
I guess I know the reason

Here in example
```
 kernel?.execute(code)?.result
      .then((result: string) => {
        setResult(result);
      })
      .catch((error: string) => {
        console.log('Error', error);
        setResult(error);
      })
      .finally(() => {
        setRunning(false);  
      }
    );
```

you are catching promise error but when executing code in `Output` here https://github.com/datalayer/jupyter-ui/blob/main/packages/react/src/components/output/OutputExecutor.ts  nobody does it and that is why I getting Runtime exception

**Comment by echarles** [comment]
OK, I see, so there is a need to better manage the exception on the promise. Would be great that you open a PR for that if it works for you.

**Comment by sok82** [comment]
@echarles Here is a PR for this - https://github.com/datalayer/jupyter-ui/pull/302

**Comment by echarles** [comment]
Merged. Closing as fixed. Thx @sok82 :+1:

**Comment by sok82** [comment]
@echarles Please publish changes so I can check it locally

**Comment by echarles** [comment]
0.18.5 is published in npm

**Comment by sok82** [comment]
Need to reopen - my solution doesn't seem to work

It breaks before we rich promise where I tried to catch 

```
Uncaught Error NameError: name 'HTML' is not defined
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[4], line 1
----> 1 display(HTML(job_satisfaction_rate_train.head(10).to_html()));

NameError: name 'HTML' is not defined
    at reject (/Users/sk/Develop/d20/d20-git-repo/d20-web-app-page-editor-ui/node_modules/@lumino/coreutils/src/promise.ts:51:1)
    at _onReply (/Users/sk/Develop/src/jupyter/kernel/KernelExecutor.ts:279:1)
    at _handleReply (/Users/sk/Develop/d20/d20-git-repo/d20-web-app-page-editor-ui/node_modules/@jupyterlab/services/src/kernel/future.ts:237:1)
    at handleMsg (/Users/sk/Develop/d20/d20-git-repo/d20-web-app-page-editor-ui/node_modules/@jupyterlab/services/src/kernel/future.ts:219:1)
    at _handleMessage (/Users/sk/Develop/d20/d20-git-repo/d20-web-app-page-editor-ui/node_modules/@jupyterlab/services/src/kernel/default.ts:1430:1)
    at <anonymous> (/Users/sk/Develop/d20/d20-git-repo/d20-web-app-page-editor-ui/node_modules/@jupyterlab/services/src/kernel/default.ts:1595:1)
    --- Promise.then ---
    at KernelConnection._onWSMessage (/Users/sk/Develop/d20/d20-git-repo/d20-web-app-page-editor-ui/node_modules/@jupyterlab/services/src/kernel/default.ts:1591:1)
```

**Comment by echarles** [comment]
@sok82 To confirm, the issue you are facing is about the Output component not displaying by default the errors, right?

**Comment by sok82** [comment]
The issue is that I  want to get rid of this in development mode

![image](https://github.com/user-attachments/assets/29856344-99fe-4556-a898-9d50891be060)

 and do not see execution error exceptions in production mode when using Output

**Comment by echarles** [comment]
running the `Outputs` example, I am not facing that issue

![Screenshot from 2024-09-09 14-24-42](https://github.com/user-attachments/assets/3a697eff-925f-4d50-9315-3147fdf4c2ec)

**Comment by sok82** [comment]
And in developer console also no sign of exception?

I tried to reproduce it with just single Output and whenever I input incorrect code and execute it - error message shows

**Comment by sok82** [comment]
@echarles Need help with this?

I just merged incoming changes for `Notebook` and `NotebookAdapter` that were made today and did nothing with it.

**Comment by sok82** [comment]
Created clear fix from current main state - https://github.com/datalayer/jupyter-ui/pull/307

@echarles Pls check

**Comment by sok82** [comment]
Hey @echarles 
Sorry to disturb you, but we need one more PR to approve with this exception handling issue. 

It didn't worked in my code AND in fact it didn't worked in examples.

I fixed this in new PR  - https://github.com/datalayer/jupyter-ui/pull/309 and updated `Output` example to show the difference.


TLDR

If you will try to execute example `Output` and run some buggy code you will get error in console about uncaught promise
![image](https://github.com/user-attachments/assets/61cfb9e3-cd4d-46d1-9e20-3757b0c6d56d)

This happens due to the fact that `lumino` wraps promises and if we do not register promise catch BEFORE execution it crashes with exception
And in case with `Output` control nobody did it 


If you will trigger buggy code execution with `Execute with error handler` button (which uses error handler) - everything plays nice and exception is handled

![image](https://github.com/user-attachments/assets/4b4e5599-3d9c-4bb0-887c-a47ba77f46b7)

 otherwise (if you press button Run) it is thrown and stays in console

**Comment by echarles** [comment]
@sok82 just merged and released 0.18.7 with that change. Thx a bunch!
